xvfb-run: clean up and update package

B4dM4n opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

#72906

###### Things done

In addition to updating the package I also renamed it from `xvfb_run` to `xvfb-run` and added a backwards compatible alias, as `xvfb_run` is not an intuitive package name to install when trying to run the `xvfb-run` script.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
<details>
  <summary>nix-review result</summary>
  
  ```log
builder for '/nix/store/5807p8q8islcjxvk7nqziw0frld2pjv4-ping-0.6.0.drv' failed with exit code 1; last 10 log lines:
                      if(tempDataList.contains(variableName)){
                         ^^^^^^^^^^^^
  ../src/Application.vala:868.25-868.36: error: The name `tempDataList' does not exist in the context of `PingApp.setupSignals._lambda34_'
                          tempDataList[variableName] = new_text;
                          ^^^^^^^^^^^^
  ../src/Application.vala:869.67-869.78: error: The name `tempDataList' does not exist in the context of `PingApp.setupSignals._lambda34_'
                          testObjs[id].data = Soup.Form.encode_hash(tempDataList);
                                                                    ^^^^^^^^^^^^
  Compilation failed: 36 error(s), 8 warning(s)
  ninja: build stopped: subcommand failed.
cannot build derivation '/nix/store/m141iwy3anw90ygsrp243kva1xlljkbp-env.drv': 1 dependencies couldn't be built
[1 built (1 failed)]
error: build of '/nix/store/m141iwy3anw90ygsrp243kva1xlljkbp-env.drv' failed
1 package failed to build:
ping

102 package were built:
adapta-gtk-theme almanah amtk autokey balsa calls chrome-gnome-shell contrast coq_8_10 denemo empathy ephemeral epiphany flent fractal gImageReader gajim girara gitg glom gnome-builder gnome-latex gnome-photos gnome-podcasts gnome-usage gnome3.anjuta gnome3.cheese gnome3.devhelp gnome3.geary gnome3.gedit gnome3.gnome-calculator gnome3.gnome-calendar gnome3.gnome-contacts gnome3.gnome-control-center gnome3.gnome-music gnome3.gnome-session gnome3.gnome-shell gnome3.gnome-terminal gnome3.gnome-tweak-tool gnome3.gpaste gnome3.gtksourceview gnome3.gtksourceview4 gnome3.gtksourceviewmm gnome3.libdazzle gnome3.meld gnome3.mutter gnome3.nemiver gnome3.pomodoro gnome3.sushi gnome3.totem gnomeExtensions.gsconnect gobby5 gscan2pdf gtkd gtkpod gtksourceviewmm4 gtranslator gupnptools i3 i3-gaps i3-layout-manager i3-wk-switch jucipp kmymoney libhandy lutris lutris-free mate.mate-applets mate.pluma nasc notejot notes-up ocamlPackages.lablgtk3-sourceview3 pantheon.elementary-code pantheon.elementary-gsettings-schemas pantheon.elementary-onboarding pantheon.elementary-session-settings pantheon.notes-up paperwork polybarFull pspp python27Packages.cartopy python27Packages.i3ipc python27Packages.py3status python37Packages.dogtail python37Packages.i3ipc python37Packages.py3status python38Packages.dogtail python38Packages.i3ipc python38Packages.py3status pytrainer quilter rednotebook sequeler sysprof tepl tilix virtmanager xfce.mousepad xpad xvfb-run zathura
 ```
</details>

- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) (`210.2M -> 210.1M`)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @davidak

B4dM4n committed:
xvfb-run: rename package

B4dM4n committed:
xvfb-run: add backward compatible alias

B4dM4n committed:
xvfb-run: cleanup package

worldofpeace commented:
Hmm, I'm not sure why this is in my memory, but I thought that `xvfb-run` actually came from an arch linux developer.

B4dM4n commented:
It looks like the first release is in Debian was in 2006 ([commit](https://salsa.debian.org/xorg-team/xserver/xorg-server/commit/76352eb081da211dbb15e90752a6b4e09e4fcc0e)). Arch Linux included the same version in 2009 via [c9f8fec3](https://git.archlinux.org/svntogit/packages.git/commit/trunk/xvfb-run?h=packages/xorg-server&id=c9f8fec3d82375d2848fa14425362e90bbb421d7) ([related issue](https://bugs.archlinux.org/task/15740)) from Debian.

The current version in Arch Linux comes from _The T2 SDE Project_ which is also the current source for this package.
Debian still maintains its own version ([history](https://salsa.debian.org/xorg-team/xserver/xorg-server/commits/debian-unstable/debian/local/xvfb-run)) actively.

Functionally should be still equivalent, but the Debian version uses a more reliable cleanup function and has an improved startup speed.

I just noticed that the Debian version uses a different license (MIT) which needs to be changed.

davidak reviewed (changes requested):
Looks good.

Only the description could be improved i think.

See what others are using: https://repology.org/project/xvfb-run/information

Thanks for your contribution!

B4dM4n committed:
xvfb-run: use release tag as source version

B4dM4n committed:
xvfb-run: update meta values

davidak commented on pkgs/tools/misc/xvfb-run/default.nix:
```diff
@@ -29,9 +29,9 @@ stdenv.mkDerivation rec {
-    inherit (src.meta) homepage;
-    license = licenses.gpl2;
+    description = "A wrapper for the Xvfb command which simplifies the task of running commands";
+    homepage = src.meta.homepage +  "debian/local";
```
It's good to be able to just read the URL there...

B4dM4n replied:
I saw the usage of `inherit (src.meta) homepage;` in some packages lately and I liked the style.

In this case, the homepage line gets noticeably shorter, from 107 (when using the full path to `debian/local`) to 74 characters.

```
    homepage = "https://salsa.debian.org/xorg-team/xserver/xorg-server/tree/debian-unstable/debian/local";
```

vs

```
    homepage = src.meta.homepage +  "/tree/debian-unstable/debian/local";
```
(current version is incomplete, will be fixed)

jtojnar replied:
Yeah. Personally, I am not a fan of `src.meta.homepage` either. I frequently click on `meta.homepage` URL when reviewing PRs and this prevents this use case without evaluation the expression.

B4dM4n commented:
> Only the description could be improved i think.

Good point, I updated the meta values to match the information found on repology.

I also updated the source revision to use the Debian release tag. This gives the version number a better meaning and makes it more relatable to the used commit.

jtojnar commented on pkgs/tools/misc/xvfb-run/default.nix:
```diff
@@ -1,29 +1,38 @@
   meta = with stdenv.lib; {
+    description = "A wrapper for the Xvfb command which simplifies the task of running commands";
+    homepage = src.meta.homepage +  "debian/local";
+    license = licenses.mit;
```
Where did you find this? Are you sure that is not just a license of Xorg? The manpage contains `gpl2Plus`.

B4dM4n replied:
Since the repository is a fork of the Xorg tree and the script itself has no license notice, it is distributed under the Xorg license, right?

The Xorg license is `a slight variant of the common MIT license form`.

The `xorg.xorgserver` package currently has no license (see https://nixos.org/nixos/packages.html?attr=xorg.xorgserver&channel=nixos-19.09) but the `xwayland` package build from the same source is declared a `licenses.mit` (see https://nixos.org/nixos/packages.html?attr=xwayland&channel=nixos-19.09), so I thought `licenses.mit` would be correct.

I didn't notice the GPL license in the manpage.

Which license is the correct one now?

davidak replied:
Since the script itself has no license information, the general license for the repo applies.

https://salsa.debian.org/xorg-team/xserver/xorg-server/blob/debian-unstable/COPYING

But the manpage is GPL ;)

Which seems messed up and unintended... maybe open an issue upstream to clearify.

jtojnar commented:
Hmm, the T2 version seems to have introduced an `--auto-display` flag.

It would be nice if we managaed to upstream the script into xorg itself.

davidak commented:
can we get this merged?
